IBPSA/Fluid/HeatPumps/ModularReversible/Controls/Safety/OnOff.mo

```diff
@@ -0,0 +1,253 @@
+  IBPSA.Fluid.HeatPumps.ModularReversible.Controls.Safety.BaseClasses.OnPastThreshold locTimCtr(
+   final minOnTime=minOffTime) if use_minOffTime
+    "Check if device should be locked"
+    annotation (Placement(transformation(extent={{20,10},{40,30}})));
+  Modelica.Blocks.Logical.Not notIsOn "=true if device is off"
+    annotation (Placement(transformation(extent={{-100,-60},{-80,-40}})));
+  IBPSA.Fluid.HeatPumps.ModularReversible.Controls.Safety.BaseClasses.OnPastThreshold runTimCtr(
+      final minOnTime=minOnTime) if use_minOnTime "Check if device needs to run"
+    annotation (Placement(transformation(extent={{0,90},{20,110}})));
+  Modelica.Blocks.Logical.And andIsAblToTurOn(
+    y(start=true, fixed=true))
+    "=false to lock the device off"
+    annotation (Placement(transformation(extent={{60,-70},{80,-50}})));
+
+  Modelica.Blocks.Sources.BooleanConstant booConstCycRat(final k=true)
+    if not use_maxCycRat "Constant value for disabled option"
+    annotation (Placement(transformation(extent={{20,-100},{40,-80}})));
+  Modelica.Blocks.Sources.BooleanConstant booConstLocTim(final k=true)
+    if not use_minOffTime "Constant value for disabled option"
+    annotation (Placement(transformation(extent={{20,-20},{40,0}})));
+  Modelica.Blocks.Sources.BooleanConstant booConstRunTim(final k=true)
+    if not use_minOnTime "Constant value for disabled option"
+    annotation (Placement(transformation(extent={{0,60},{20,80}})));
+  Modelica.Blocks.Logical.Not notSetOn "Device is not set to turn on"
+    annotation (Placement(transformation(extent={{-100,18},{-80,38}})));
+  Modelica.Blocks.Logical.And andTurOff(
+    y(start=not preYSet_start, fixed=true))
+    "Check if device is on and is set to be turned off"
+    annotation (Placement(transformation(extent={{-40,0},{-20,20}})));
+  Modelica.Blocks.Logical.And andTurOn(
+    y(start=preYSet_start, fixed=true))
+    "Check if device is Off and is set to be turned on"
+    annotation (Placement(transformation(extent={{-40,-100},{-20,-80}})));
+
+  Modelica.Blocks.Logical.And andStaOn
+    "=true if the device is on and wants to stay on"
+    annotation (Placement(transformation(extent={{-40,40},{-20,60}})));
+  Modelica.Blocks.Logical.And andStaOff
+    "=true if the device is off and wants to stay off"
+    annotation (Placement(transformation(extent={{-40,-40},{-20,-20}})));
+  Modelica.Blocks.Nonlinear.Limiter lim(uMax=1, uMin=ySetRed) "Keep device off"
+    annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={-90,110})));
+
+protected
+  Integer devRunMin(start=0, fixed=true)
+    "Indicates if device needs to run at minimal limit";
+  Integer devTurOff(start=0, fixed=true)
+    "Indicates if device needs to turn off";
+  Integer devNorOpe(start=1, fixed=true)
+    "Indicates if device is at normal operation";
+equation
+  yOut = ySet * devNorOpe + 0 * devTurOff +ySetRed  * devRunMin;
```

Comment:
It turns out that this model does not translate in Optimica, see the comment https://github.com/ibpsa/modelica-ibpsa/pull/1628#issuecomment-1841485335
There is also this suspicious line that contains `yOut = .... + 0 * devTurOff + ....` and the model mixes connect statements with assignment. I tried to remove the `pre` that is applied to an input that this does not help. 
Can you move the assignment into its own block, and instantiate this block here. Maybe that gives finer granularity in isolating the singularity that Optimica finds.

Reply:
This is not an issue anymore, or is it?